c10/core/DeviceType.h

```diff
@@ -50,15 +50,18 @@ enum class DeviceType : int8_t {
   Lazy = 17, // Lazy Tensors
   IPU = 18, // Graphcore IPU
   PrivateUse1 = 19, // PrivateUse1 device
+  DML = 20, // DirectML / Microsoft
```

Comment:
Hmm, should PrivateUse1 always be the last one in the list? (not sure to be frank..)

Reply:
For this list it's OK for it not to be last. In BackendComponent, though, it's important for the private use keys to be last.